### PR TITLE
Fix build node and docker-image

### PR DESCRIPTION
**Pull request by im-kulikov**
- Make Dockerfile useful and minimal
- Update Makefile commands build / build-image
- Update Readme: fix Dep-manager and minimal requirements
- Docker should not ignore vendor (go build -mod=vendor)

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,16 +1,40 @@
...
+COPY --from=builder /go/bin/node     /usr/bin/node
+COPY --from=builder /etc/ssl/certs/ca-certificates.crt /etc/ssl/certs/
 
-RUN chmod u+x /usr/bin/neo-go
-RUN mkdir -p /chains
+ENTRYPOINT ["/usr/bin/node"]
 
-ENTRYPOINT ["neo-go"]
-CMD ["node", "--config-path", "./config", "--testnet"]
+CMD ["node", "--config-path", "./config", "--testnet"]
```
Comment (roman-khimov) [inline Dockerfile]: "No newline at the end of file", hmm...

File: README.md [hunk 2]
```diff
@@ -31,9 +31,11 @@ A complete toolkit for the NEO blockchain, including:
 
 ## Installation
 
+Go: 1.12+
```
Comment (roman-khimov) [inline README.md]: Not really, we even have 1.11 configured in CircleCI. Also, it has no relation to docker again.

File: go.mod [hunk 3]
```diff
@@ -1,5 +1,7 @@
 module github.com/CityOfZion/neo-go
 
+go 1.12
```
Comment (roman-khimov) [inline go.mod]: NACK.

File: Makefile [hunk 4]
```diff
@@ -1,19 +1,24 @@
...
 BUILD_TIME = "$(shell date -u +\"%Y-%m-%dT%H:%M:%SZ\")"
-VERSION = $(shell cat ./VERSION)
 REPONAME = "neo-go"
 NETMODE ?= "privnet"
 
-build:
-	@echo "=> Building darwin binary"
-	@go build -i -ldflags "-X github.com/CityOfZion/neo-go/config.Version=${VERSION}-dev -X github.com/CityOfZion/neo-go/config.BuildTime=${BUILD_TIME}" -o ./bin/neo-go ./cli/main.go
+REPO ?= `go list -m`
+VERSION ?= `git describe --tags 2>/dev/null || git rev-parse --short HEAD`
```
Comment (roman-khimov) [inline Makefile]: That looks like a separate issue, not related to docker itself, thus at least deserving its own commit. Also, to make it really useful we would need some `vX.Y.Z-dev` tags.

Comment (im-kulikov) [reply Makefile]: - Before `VERSION = $(shell cat ./VERSION)`
- Now `VERSION ?= $(git describe --tags 2>/dev/null || git rev-parse --short HEAD)`

there's no `./VERSION` file in repo and build fails..  needs pass version into docker-build

**Comment by roman-khimov** [comment]
BTW, I think this one should also reference #311 because IIUC it fixes it.

File: cli/main.go [hunk 5]
```diff
@@ -7,12 +7,14 @@ import (
...
 	"github.com/CityOfZion/neo-go/cli/vm"
 	"github.com/CityOfZion/neo-go/cli/wallet"
+	"github.com/CityOfZion/neo-go/config"
 	"github.com/urfave/cli"
 )
 
 func main() {
 	ctl := cli.NewApp()
 	ctl.Name = "neo-go"
+	ctl.Version = config.Version
```
Comment (roman-khimov) [inline cli/main.go]: I'm wondering if it fixes #102? If so, it should be mentioned in the commit message.

File: Dockerfile (hunk 1 above)
[the remark this answers is not in the document]
Comment (roman-khimov) [reply Dockerfile]: Still there.

File: cli/main.go (hunk 5 above)
Comment (im-kulikov) [reply cli/main.go]: done

**Pull request merged**